internal/engine/upper.go

```diff
@@ -224,6 +224,94 @@ func handleBuildStarted(ctx context.Context, state *store.EngineState, action bu
 	removeFromTriggerQueue(state, mn)
 }
 
+func updateBuildStatus(status *store.BuildStatus, br model.BuildRecord, result store.BuildResult, isBuildSuccess bool) {
+	// Record the last result
```

Comment:
```suggestion
	// Record the result
```
(otherwise it seems like you're talking about the result BEFORE the one you're currently processing)